packages/frontend/core/src/modules/workbench/view/split-view/split-view.tsx

```diff
@@ -102,6 +106,10 @@ export const SplitView = ({
         const fromIndex = views.findIndex(v => v.id === active.id);
         const toIndex = views.findIndex(v => v.id === over?.id);
         onMove?.(fromIndex, toIndex);
+        setVisibility(false);
+        setTimeout(() => {
```

Comment:
it's a potential hook update after unmount, you could do it with an `useEffect`

```ts
useEffect(() => {
  if (!visible) {
    setTimeout(() => {
      setVisible(true)
    }, 0)

    return clear
  }
}, [visible])
```

Reply:
or directly in render, it will be rendered in next update, I think
```ts
if (!visible) {
  setVisible(true)
}
```

Reply:
I think this case is extremely rare, may not worth it adding an useEffect.
The following does not work. I think setting state in render will reschedule a sync render in the same tick.

```
if (!visible) {
  setVisible(true)
}
```

Reply:
rare but it's still not correct. it will cost time debugging if it happens one day especially when you forget about these lines of code